src/components/tiles/table/table-toolbar.tsx

```diff
@@ -30,22 +33,30 @@ export const TableToolbar: React.FC<IProps> = observer(({
                     ...others
                   });
 
-  const buttonSettings = useSettingFromStores("tools", "table") as unknown as string[] | undefined;
+  const buttonSettings = useSettingFromStores("tools", "table") as unknown as IButtonSetting[] | undefined;
   const buttons = buttonSettings || defaultButtons;
 
-  const getToolbarButton = (toolName: string) => {
-    switch (toolName) {
-      case "set-expression":
-        return <SetExpressionButton key={toolName} onClick={onSetExpression} />;
-      case "delete":
-        return <DeleteSelectedButton key={toolName} onClick={deleteSelected} />;
-      case "link-tile":
-        return <LinkTileButton
-                 key={toolName}
-                 isEnabled={isLinkEnabled}
-                 getLinkIndex={getLinkIndex}
-                 onClick={showLinkDialog}
-               />;
+  const getToolbarButton = (toolName: IButtonSetting) => {
+    if (typeof toolName === "string") {
+      switch (toolName) {
+        case "set-expression":
+          return <SetExpressionButton key={toolName} onClick={onSetExpression} />;
+        case "delete":
+          return <DeleteSelectedButton key={toolName} onClick={deleteSelected} />;
+        case "link-tile":
+          return <LinkTileButton
+                  key={toolName}
+                  isEnabled={isLinkEnabled}
+                  getLinkIndex={getLinkIndex}
+                  onClick={showLinkDialog}
+                />;
+      }
+    } else {
+      const realToolName = toolName[0];
```

Comment:
```suggestion
      // if `toolName` is an array, it contains the type as well
      const [realToolName, tileType] = toolName;
```
Or some other suitable comment, since the code is a bit opaque otherwise.